trio/_util.py

```diff
@@ -133,8 +133,10 @@ def __init__(self, func, args, kwds):
                 # fixes the impedance mismatch between the throw() protocol
                 # and the __exit__() protocol.
                 #
-                if sys.exc_info()[1] is not value:
-                    raise
+                if sys.exc_info()[1] is value:
+                    return False
+                raise
+            raise RuntimeError("generator didn't stop after throw()")
```

Comment:
This last line (`raise RuntimeError(...)`) is redundant -- we there's another check for the same thing up above. Should be deleted.